### DISCUSSION TITLE
Get rid of ENV file, include in docker compose.

### DISCUSSION THREAD

**Discussion by rekcodocker** [immich-app/immich#9037] [Feature Request]
### The bug

The . env file requires extra actions. 
This is harder in a managed docker environment where docker compose is run by management software. 
Also the env-file does not contain anything unique that can not be in the docker compose.

### The OS that Immich Server is running on

Debian latest

### Version of Immich Server

immich latest

### Version of Immich Mobile App

immich latest

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
The default unaltered yml.
```

### Your .env content

```Shell
The default unaltered
```

### Reproduction steps

```bash
1. Install it with docker compose according to the instructions
2. note that the ENV file is hard to get when you use a docker manager like portainer.
3. File a bug/feature request at gihub
...
```

### Additional information

The env file seems to contain nothing that can't be in the compose-file.

**Comment by bo0tzz** [comment]
The env file lets us template things inside the compose file, such as UPLOAD_LOCATION. It also helps us avoid duplicating the same var under multiple services, for example for the database connection.

**Reply by rekcodocker** [reply]
Ok, I got it to work, I'll close this.

**Reply by jrasm91** [reply]
To be fair, I'm actually heavily in favor of this. It is more clear and obvious what is used where and how. Plus you can use yaml anchors to share the duplicated variables between immich-server and immich-microservices.

**Reply by mmomjian** [reply]
It would work fine for everything other than Postgres variables. That would require some variables to be renamed. (DB_PASSWORD vs POSTGRES_PASSWORD)

**Reply by bo0tzz** [reply]
Personally I like the clean separation between the two files. It keeps the compose file straightforward, and clearly shows the place that people should change things (in the .env).
